data-node/src/test/java/org/graylog/datanode/integration/DatanodeSecurityTestUtils.java

```diff
@@ -0,0 +1,100 @@
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.datanode.integration;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.graylog.datanode.configuration.variants.KeystoreInformation;
+import org.graylog.security.certutil.CertutilCa;
+import org.graylog.security.certutil.CertutilCert;
+import org.graylog.security.certutil.CertutilHttp;
+import org.graylog.security.certutil.console.TestableConsole;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Enumeration;
+
+public class DatanodeSecurityTestUtils {
+    public static KeyStore buildTruststore(KeystoreInformation ca) throws IOException, KeyStoreException, CertificateException, NoSuchAlgorithmException {
+        try (FileInputStream fis = new FileInputStream(ca.location().toFile())) {
+
+            KeyStore caKeystore = KeyStore.getInstance("PKCS12");
+            caKeystore.load(fis, ca.password());
+
+            KeyStore trustStore = KeyStore.getInstance("PKCS12");
+            trustStore.load(null, null);
+
+            final Enumeration<String> aliases = caKeystore.aliases();
+            while (aliases.hasMoreElements()) {
+                final String alias = aliases.nextElement();
+                final Certificate cert = caKeystore.getCertificate(alias);
+                if (cert instanceof final X509Certificate x509Certificate) {
+                    trustStore.setCertificateEntry(alias, x509Certificate);
+                }
+            }
+            return trustStore;
+        }
+    }
+
+    public static KeystoreInformation generateCa(Path dir) {
+        final Path certPath = dir.resolve("test-ca.p12");
+        final String password = RandomStringUtils.randomAlphabetic(10);
+        final TestableConsole input = TestableConsole.empty().silent()
+                .register("Enter CA password", password);
+        final CertutilCa command = new CertutilCa(certPath.toAbsolutePath().toString(), input);
+        command.run();
+        return new KeystoreInformation(certPath, password);
+    }
+
+    public static KeystoreInformation generateTransportCert(Path dir, KeystoreInformation ca, String... containerHostnames) {
+        final Path transportPath = dir.resolve("transport-" + RandomStringUtils.randomAlphabetic(10) + ".p12");
+        final String password = RandomStringUtils.randomAlphabetic(10);
+        TestableConsole inputCert = TestableConsole.empty().silent()
+                .register("Enter CA password", ca.passwordAsString())
```

Comment:
Those magic strings like "Enter CA password" appear in many place in the code. I wonder if they do not deserve a constant or another mechanism that will allow to modify them in one place only, whenever they need modification.

Reply:
Yeah, these user inputs would probably deserve an enum, expressing the intention behind the user-facing prompt. Let me see if I can suggest something nicer.

Reply:
Ok, this will deserve its own PR, it will be a slightly bigger refactoring.